tests/test_formats.py

```diff
@@ -346,85 +338,149 @@ def test_RootFile(self):
-                                                      consistent_snapshot,
-                                                      compression_algorithms)))
-    metadata = make_metadata(version, expires, keydict, roledict,
-                             consistent_snapshot, compression_algorithms)
-    self.assertTrue(isinstance(from_metadata(metadata), tuf.formats.RootFile))
+    self.assertTrue(tuf.formats.ROOT_SCHEMA.matches(
+        tuf.formats.build_dict_conforming_to_schema(
+        tuf.formats.ROOT_SCHEMA,
+        _type='Root',
+        version=version,
+        expires=expires,
+        keys=keydict,
+        roles=roledict,
+        consistent_snapshot=consistent_snapshot,
+        compression_algorithms=compression_algorithms)))
 
-    # Test conditions for invalid arguments.
-    bad_version = '8'
-    bad_expires = 'eight'
+
+    # Additional test arguments for invalid Root creation.
     bad_keydict = 123
     bad_roledict = 123
-    bad_compression_algorithms = ['nozip']
-
-    self.assertRaises(tuf.FormatError, make_metadata, bad_version,
-                                                      expires,
-                                                      keydict, roledict,
-                                                      consistent_snapshot,
-                                                      compression_algorithms)
-    self.assertRaises(tuf.FormatError, make_metadata, version,
-                                                      bad_expires,
-                                                      keydict, roledict,
-                                                      consistent_snapshot,
-                                                      compression_algorithms)
-    self.assertRaises(tuf.FormatError, make_metadata, version,
-                                                      expires,
-                                                      bad_keydict, roledict,
-                                                      consistent_snapshot,
-                                                      compression_algorithms)
-    self.assertRaises(tuf.FormatError, make_metadata, version,
-                                                      expires,
-                                                      keydict, bad_roledict,
-                                                      consistent_snapshot,
-                                                      compression_algorithms)
-
-    self.assertRaises(tuf.FormatError, from_metadata, 'bad')
-
-
-
-  def test_SnapshotFile(self):
-    # Test conditions for valid instances of 'tuf.formats.SnapshotFile'.
-    version = 8
-    expires = '1985-10-21T13:20:00Z'
+    bad_compression_algorithms = 5
+
+    # TODO: Later on, write a test looper that takes pairs of key-value args
+    #       to substitute in on each run to shorten this.... There's a lot of
+    #       test code that looks like this, and it'd be easier to use a looper.
```

Comment:
Agreed! :) 
I recently started to do a lot more table-driven testing (see  [in-toto/test_gpg.py](https://github.com/in-toto/in-toto/blob/v0.3.0/tests/test_gpg.py#L139-L166) for an example).

Reply:
It's good stuff.  I have some stuff like that in Uptane, in test_secondary.py for example.  Can't remember where else.